### ISSUE TITLE
`no-implicit-any-catch` should respect `Error` type

### ISSUE THREAD

**Issue by artfulsage** [cartant/eslint-plugin-rxjs#87]
Hey! Sorry, I don't understand why it's incorrect to define an error type as `Error` according to the rule `no-implicit-any-catch`. Shouldn't it be changed?

I think this code is correct in types universe.

```ts
throwError(() => new Error("Kaboom!")).pipe(
  catchError((error: Error) => console.error(error.message))
);
```

**Comment by cartant** [comment]
`throwError` doesn't specify a type for the error. It only specifies a type for the element.

TypeScript - and JavaScript - have no mechanism to constrain what can be thrown from called code, so there is no guarantee that what's caught - the same issue applies to `catch` statements - will be an `Error`. Code within the pipeline - or, with `catch` statements, within a `try` block - could throw anything.

In TypeScript, `catch` variables are [best typed as `unknown`](https://devblogs.microsoft.com/typescript/announcing-typescript-4-4/#use-unknown-catch-variables) because that's really what they are. This rule does the same thing for error handlers. And for the same reason.

Usually, things are more complicated that the simple example you listed above, but even with that snippet you cannot be sure that what's caught will be an `Error` instance. What if the environment it runs in has had this done to it?

```js
global.Error = class Error { constructor() { throw 42 } }
```

**Comment by artfulsage** [comment]
@cartant I see your point and it looks reasonable in general. Thank you for your answer.

**Comment by maxime4000** [comment]
Tbh, I don't see why this rule exist if you cannot have implicit any, nor explicit type... 
How is this rule helpful if you know what you will receive 99.99% of the time, but you are force to set unknown? 
I'm typing the code so I can know what object I'm expecting, but if the rule says I cannot type this object the way I want and it doesn't want you to use `any`, why I'm using this rule in the beginning?
Great Typescript project are those where you get to know every object you are expecting.
IMO, a rule that doesn't allow you to implicit any, but also doesn't allow you to explicit type your object isn't a useful rule.
My choice is `unknown` or explicit any, which is the same as not typing the error. I was expecting error should be type or set to unknown, but typing the error isn't allowed...

Why is this not allow?
```.ts
import { HttpErrorResponse } from '@angular/common/http';
(error: HttpErrorResponse) => {}
```
Unless you can explain me why I should keep this rule, this one shouldn't exist and goes "off" in my book.
